features/steps/oauth.py

```diff
@@ -92,6 +105,90 @@ def step_impl(context):
     context.authorization_received = response
 
 
+@when('I ask for authorization')
+def step_impl(context):
+    context.code = context.oauth.request_authorization()
+
+
+@when('I exchange my grant code')
```

Comment:
Nit: Official spec language for this is "authorization code" or "authorization code grant" rather than "grant code"